### DISCUSSION TITLE
Interest in Industrial meter reading with OpenVINO[GSOC]

### DISCUSSION THREAD

**Discussion by ryan-utopia** [openvinotoolkit/openvino#16227] [Google Summer of Code]
Hi, OpenVINO developers,

I'm Ryan Chen, a sophomore undergraduate student majoring in Artificial Intelligence at Beijing Normal University. Despite being a sophomore, I possess exceptional practical skills. I am a passionate enthusiast of artificial intelligence and have been following its development in various aspects for the past few years. 

In college, I have learned a lot of foundational knowledge related to artificial intelligence and gradually mastered the process and methods of constructing artificial intelligence. In the past year, I also received a first-class scholarship from our university. I have systematically studied machine learning in school and have experience in data science competitions. I have been widely studying in the field of machine learning and also paying attention to deep learning.

During the summer of my freshman year, I interned at the Openeuler community, helping to migrate the ROS-based SLAM algorithm SVO (Semi-Direct Monocular Visual Odometry). In school, I have also done some vision-related projects based on esp32, such as face recognition, security detection, and other IoT application constructions. This time, when I checked out the GSOC organizations, I was delighted to discover such an AI community and its goal is to deploy AI models, and I also hope to contribute my own strength.

I am very excited to make a contribution to OpenVINO this summer. I am particularly interested in these two projects:

1. Industrial meter reading with OpenVINO
2. Showcase performance of PyTorch Image Models (timm) with OpenVINO

Especially the project Industrial meter reading with OpenVINO, which is coincide with my hope to work in the industrial field AI in the future and liberate more human labor.

Before starting, I would like to ask if there are any additional prerequisites for these projects besides the prerequisites already posted on the OpenVINO wiki page. I am enthusiastic about these projects and hope to discuss them further, or maybe just give me some small tasks related to the project? Thank you for your valuable time and consideration, and I look forward to hearing from you soon.

**Comment by adrianboguszewski** [comment]
@OpenVINO-dev-contest @zhuo-yoyowz, this is related to your project :)

**Comment by openvino-dev-samples** [comment]
Hi @JacketChenlll Thanks for your passion on industrial AI and OpenVINO's contribution. I think we can start from our [Prerequisite task](https://github.com/openvinotoolkit/openvino/wiki/Google-Summer-Of-Code#prerequisite-task). Meanwhile, the [project proposal](https://github.com/openvinotoolkit/openvino/wiki/Google-Summer-Of-Code#application-template) can be prepared as well.

**Reply by ryan-utopia** [reply]
Thank you so much! I'm working on the [[Prerequisite task](https://github.com/openvinotoolkit/openvino/wiki/Google-Summer-Of-Code#prerequisite-task)](https://github.com/openvinotoolkit/openvino/wiki/Google-Summer-Of-Code#prerequisite-task) and writing my proposal for this project these days, hope to talk more deeply with you soon.

**Comment by abangwal** [comment]
Hii @OpenVINO-dev-contest,

I am interested in contributing to the Industrial meter reading idea and completed my [prerequisite task](https://github.com/openvinotoolkit/openvino_notebooks/pull/871).
While writing the proposal I can't comprehend what is meant by "solution using DL framework" in the idea description.
Are we supposed to create a whole meter-detection and reading model in the DL frameworks or just implement the already available solution [here](https://github.com/openvinotoolkit/openvino_notebooks/tree/main/notebooks/203-meter-reader) in DL frameworks with demos for open_model_zoo and openvino_notebooks?

Regards,
ashish

**Reply by openvino-dev-samples** [reply]
> Hii @OpenVINO-dev-contest,
> 
> I am interested in contributing to the Industrial meter reading idea and completed my [prerequisite task](https://github.com/openvinotoolkit/openvino_notebooks/pull/871). While writing the proposal I can't comprehend what is meant by "solution using DL framework" in the idea description. Are we supposed to create a whole meter-detection and reading model in the DL frameworks or just implement the already available solution [here](https://github.com/openvinotoolkit/openvino_notebooks/tree/main/notebooks/203-meter-reader) in DL frameworks with demos for open_model_zoo and openvino_notebooks?
> 
> Regards, ashish

hi @ashish-2005 , thanks for your interests on our program. Since the current meter reader notebook is in PaddlePaddle framework, we would like to build a new End2End(from training to inference) solution with Pytorch or Tensorflow. In phase one, I think it should be a demo package, including a updated 203 notebook for openvino inference with both Paddle, and Pytorch/TF pre-trained model, a training script/instruction, and datasets (optional, you can collect them yourself or reuse a free datasets from Paddle). Does it make sense ?

**Reply by abangwal** [reply]
thank you for the response @OpenVINO-dev-contest ,

Now I got the main idea and will update my proposal accordingly.
can you please share the email where I can send my proposal for further discussion if it's okay with you?

regards
ashish

**Reply by abangwal** [reply]
Hii @OpenVINO-dev-contest 
I am creating a rough solution, for which I need data. Can you please share the free dataset from Paddle ? it would be a great help.
Thanks.

**Reply by openvino-dev-samples** [reply]
Yes You can download them through the link below:

[**Test**](https://bj.bcebos.com/paddlex/examples/meter_reader/datasets/meter_test.tar.gz)
[**Detection**](https://bj.bcebos.com/paddlex/examples/meter_reader/datasets/meter_det.tar.gz)
[**Segmentation**](https://bj.bcebos.com/paddlex/examples/meter_reader/datasets/meter_seg.tar.gz)

**Comment by openvino-dev-samples** [comment]
@adrianboguszewski Pls help to provide a space for proposal submission. thanks

**Reply by adrianboguszewski** [reply]
The final application must be uploaded via the GSoC web app. Otherwise, it won't be considered by Google. However, if we talk about drafts, you can use an email or any other way which provides enough confidentiality.

**Comment by ryan-utopia** [comment]
@OpenVINO-dev-contest ，I have completed the submission of my pre-task PR (may need further modifications) and submitted my proposal. If you have the chance to review my proposal, please feel free to provide any feedback. Thanks.

**Comment by abangwal** [comment]
Hii @OpenVINO-dev-contest,

I have submitted my proposal, can you please review it and share places for improvement? it would be appreciated.

Thanks
